src/map/utils/synthutils.cpp

```diff
@@ -659,77 +660,83 @@ namespace synthutils
             {
-                nextSlotID = PChar->CraftContainer->getInvSlotID(slotID + 1);
-            }
-
-            random = xirand::GetRandomNumber(1.);
+                if (slotID != 8)
+                {
+                    nextSlotID = PChar->CraftContainer->getInvSlotID(slotID + 1);
+                }
 
-            if (random < lostItem)
-            {
-                PChar->CraftContainer->setQuantity(slotID, 0);
-                lostCount++;
-            }
-            totalCount++;
+                random = xirand::GetRandomNumber(1.);
 
-            if (invSlotID != nextSlotID)
-            {
-                CItem* PItem = PChar->getStorage(LOC_INVENTORY)->GetItem(invSlotID);
+                if (random < lostItem)
+                {
+                    PChar->CraftContainer->setQuantity(slotID, 0);
+                    lostCount++;
+                }
+                totalCount++;
 
-                if (PItem != nullptr)
+                if (invSlotID != nextSlotID)
                 {
-                    PItem->setSubType(ITEM_UNLOCKED);
-                    PItem->setReserve(PItem->getReserve() - totalCount);
-                    totalCount = 0;
+                    CItem* PItem = PChar->getStorage(LOC_INVENTORY)->GetItem(invSlotID);
 
-                    if (lostCount > 0)
-                    {
-                        charutils::UpdateItem(PChar, LOC_INVENTORY, invSlotID, -(int32)lostCount);
-                        lostCount = 0;
-                    }
-                    else
+                    if (PItem != nullptr)
                     {
-                        PChar->pushPacket(new CInventoryAssignPacket(PItem, INV_NORMAL));
+                        PItem->setSubType(ITEM_UNLOCKED);
+                        PItem->setReserve(PItem->getReserve() - totalCount);
+                        totalCount = 0;
+
+                        if (lostCount > 0)
+                        {
+                            charutils::UpdateItem(PChar, LOC_INVENTORY, invSlotID, -(int32)lostCount);
+                            lostCount = 0;
+                        }
+                        else
+                        {
+                            PChar->pushPacket(new CInventoryAssignPacket(PItem, INV_NORMAL));
+                        }
                     }
+                    invSlotID = nextSlotID;
+                }
+                nextSlotID = 0;
+                if (invSlotID == 0xFF)
+                {
+                    break;
                 }
-                invSlotID = nextSlotID;
-            }
-            nextSlotID = 0;
-            if (invSlotID == 0xFF)
-            {
-                break;
             }
         }
 
-        if (PChar->loc.zone->GetID() != 255 && PChar->loc.zone->GetID() != 0)
+        // Push "Synthesis failed" messages.
+        if (PChar->loc.zone->GetID() != ZONE_MONORAIL_PRE_RELEASE &&
+            PChar->loc.zone->GetID() != ZONE_ABYSSEA_EMPYREAL_PARADOX) // Why?
```

Comment:
`MAX_ZONEID                          = 300,`

max used to be 255, prolly how that got there